src/main/java/org/squiddev/cobalt/function/LuaInterpreter.java

```diff
@@ -640,6 +655,850 @@ A C R(A+3), ... ,R(A+2+C):= R(A)(R(A+1),
 		}
 	}
 
+	private static Function<UnwindableRunnable, UnwindableCallable<EvalCont>> continuation(LuaState state, int pc, Prototype proto) {
+		final DebugHandler handler = state.debug;
+		final int instr = proto.code[pc] >> POS_OP & MAX_OP;
+
+		return f -> {
+			final UnwindableCallable<EvalCont> callable = di -> {
+				final DebugState ds = DebugHandler.getDebugState(state);
```

Comment:
Can we pass this as a parameter to the `UnwindCallable`? It'd be nice if we could avoid calling this each time (even if it's just a pair of field accesses). Don't know...

Reply:
Absolutely, this is one of the changes I want to make in a refactor of `EvalCont`. I'd like to try passing an `EvalCont` as the lambda parameter and keep cached information in it, editing fields like `programCounter` in-place rather than returning a new object. It's one of the changes that have to be profiled to verify that they actually help.

Reply:
I briefly tried to pass the debug state in `EvalCont` after the refactor, which resulted in noticeably worse performance in `PerformanceTest`. I didn't fully profile the issue though.

Reply:
> That's absurd. Why are JITs so terrible :D:.

I stand by this statement. Lovely.

Reply:
I guess we could have it as a field on `EvalCont`. But given that the instruction hooks are done in the main function, I guess it's less of an issue now.